### PR TITLE
Remove util deprecated functions.

### LINKED ISSUE #2658: Remove deprecated function iris.util.broadcast_weights
The function `iris.util.broadcast_weights` was deprecated in Iris 1.6

Replacement: use `iris.util.broadcast_to_shape`

https://github.com/SciTools/iris/blob/master/lib/iris/util.py#L46

### LINKED ISSUE #2659: Remove deprecated function iris.util.ensure_array
The function `iris.util.ensure_array` was deprecated in 1.7

https://github.com/SciTools/iris/blob/master/lib/iris/util.py#L976

### LINKED ISSUE #2660: Remove deprecated class iris.util._Timers
The class `iris.util._Timers` was deprecated in Iris 1.7

https://github.com/SciTools/iris/blob/master/lib/iris/util.py#L985

### PR DESCRIPTION
**Pull request by lbdreyer**
Fixes #2660 
Fixes #2658 
Fixes #2659

### CODE REVIEW

**Comment by lbdreyer** [comment]
There was no testing for these iris.util functions that I needed to remove which is interesting

File: lib/iris/util.py [hunk 1]
```diff
@@ -43,52 +43,6 @@
...
-        in *weights* should be the same as the relative ordering of the
-        corresponding dimensions in *array*. For example, if *array* is
-        4d with dimensions (ntime, nlev, nlat, nlon) and *weights*
-        provides latitude-longitude grid weightings then *dims* could be
-        set to [2, 3] or [3, 2] but *weights* must have shape
-        (nlat, nlon) since the latitude dimension comes before the
-        longitude dimension in *array*.
-
-    """
-    warn_deprecated('broadcast_weights() is deprecated and will be removed '
```
Comment (dkillick) [inline lib/iris/util.py]: @lbdreyer you will also need to remove the import for this function on [L41](https://github.com/SciTools/iris/pull/2686/files#diff-b6f4583aa861d9305fec9d55d9c18dfaL41).

Comment (lbdreyer) [inline lib/iris/util.py]: Good spot!

**Comment by lbdreyer** [comment]
@dkillick I have now removed the unused import

**Comment by dkillick** [comment]
Note: merging this also fixed #1591 and duplicated/replaced/closed #1594.

**Pull request merged**